plugins/github-deployments/src/api/index.ts

```diff
@@ -13,9 +13,43 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { parseLocationReference } from '@backstage/catalog-model';
 import { createApiRef, OAuthApi } from '@backstage/core';
+import { InputError } from '@backstage/errors';
+import { ScmIntegrationRegistry } from '@backstage/integration';
 import { graphql } from '@octokit/graphql';
 
+const getBaseUrl = (
+  scmIntegrationsApi: ScmIntegrationRegistry,
+  locations: string[],
+): string => {
+  const targets = locations
+    .map(parseLocationReference)
+    .filter(location => location.type === 'url')
+    .map(location => location.target);
+
+  if (targets.length === 0) {
+    return 'https://api.github.com';
+  }
+
+  const location = targets[0];
+  const config = scmIntegrationsApi.github.byUrl(location);
```

Comment:
Yeah as mentioned below, i wonder if we could instead make this byHost and making the host an optional parameter of the api call. The integrations actually always have a `github.com` host entry as well.